src/generic/codec.rs

```diff
@@ -239,11 +241,11 @@ where T: Decoder,
                 0 => false,
                 1 => {
                     trace!("message compressed, compression not supported yet");
-                    return Err(Status::UNIMPLEMENTED);
+                    return Err(::Error::Protocol(ProtocolError::Unsupported("compression not supported yet")));
                 },
                 _ => {
                     trace!("unexpected compression flag");
-                    return Err(Status::UNKNOWN);
+                    return Err(::Error::Protocol(ProtocolError::Unsupported("unexpected compression flag")));
```

Comment:
This is usually just plain wrong, right? Not something we don't support yet...

Reply:
I'm actually not too sure. I'll pull it up a level, but it might not be something that the caller would want to distinguish?